### ISSUE TITLE
ScraperAPI

### ISSUE THREAD

**Issue by Sikerdebaard** [scholarly-python-package/scholarly#261]
Hi folks!  

Not sure if tickets like this are appreciated. If not feel free to remove it, no hard feelings.  

I've written a quick and dirty hack to enable scholarly to use [ScraperAPI](https://ScraperAPI.com). Here's the code.

```python3
from scraper_api import ScraperAPIClient
from scholarly import scholarly, ProxyGenerator

class ScraperAPI(ProxyGenerator):
    def __init__(self, api_key):
        self._api_key = api_key
        self._client = ScraperAPIClient(api_key)
        
        assert api_key is not None
        
        super(ScrapeAPI, self).__init__()
        
        self._TIMEOUT = 120
        self._session = self._client
        self._session.proxies = {}
        
    def _new_session(self):
        self.got_403 = False
        return self._session
    
    def _close_session(self):
        pass  # no need to close the ScraperAPI client

pg = ScraperAPI('<YOUR API KEY HERE>')
scholarly.use_proxy(pg)
scholarly.set_timeout(120)

pubs = next(scholarly.search_pubs('radiomics'))
scholarly.pprint(pubs)```

**Comment by ipeirotis** [comment]
By all means. Create a pull request against the "develop" branch and we will incorporate the feature in the upcoming versions.

**Comment by MoritzImendoerffer** [comment]
Thank you for this solution. I tried it but it did not work because I got blocked by G. immediately. I try to explain what I did. Maybe you can help me in finding the cause.

1.) I used your class as you mentioned it. I got blocked immediately. This causes a `StopIteration` error because the iterator does not find the desired strings in the response (which is an HTML message which says that G. thinks the request was made by a bot).
2.) I added custom headers in the file _navigator.py in the function `get_page` just before the call of `self._session.get`:

```
                header = random.choice(HEADER_LIST)  # HEADER_LIST is a list with custom headers at the beginning of the file
                self._session.headers.update(header)
```
Which did not work for the same reason as mentioned in point 1)

3.) I stopped using your proposed solution and used my hacky solution which works for me:
in `_navigator.py:

- ` set `self._TIMEOUT = 60`
-  I added following lines right after `time.sleep` in the function `get_page`

```
                header = random.choice(HEADER_LIST)
                self._session.headers.update(header)
                payload = {'api_key': 'xxxxxx',
                           'url': pagerequest}
                # resp = self._session.get(pagerequest, timeout=timeout)
                resp = self._session.get('http://api.scraperapi.com', params=payload, timeout=timeout)
```
Now I can exhaust the iterator of the search result again and everything works fine. I know this is not a desired solution and far from being elegant. 

Any ideas why the nice solution from @Sikerdebaard is not working for me but the hack does work?

**Comment by ipeirotis** [comment]
Included in v1.2.0
